data/console/test_openssl_nodejs.sh

```diff
@@ -0,0 +1,151 @@
+    quilt setup "../SPECS/nodejs$VERSION.spec"
+    pushd "$SOURCE_DIR"
+      quilt push -a
+      # Run all test-crypto and test-tls available
+      for f in $(find test \( -path \*sequential\* -or -path \*parallel\* \) \( -name test-crypto-\* -or -name test-tls-\* \)); do
+        test_result_wrapper $VERSION $f
+      done
+    popd
+  popd
+
+  # Cleanup sources
+  rm -rf /usr/src/packages/SOURCES/*
+  rm -rf /usr/src/packages/SPECS/* 
+}
+
+main(){
+
+  OS_VERSION="$1"
+  echo "OS_VERSION: $OS_VERSION"
+
+  # Install dependencies to apply source patches and run tests
+  zypper -n in quilt rpm-build openssl-1_1
+
+  # Get list of each nodejs version found from default repo
+  local NODE_VERSIONS=$(zypper -n search nodejs | egrep -i 'nodejs[0-9]{1,2} ' | cut -d'|' -f 2 | tr -d ' '| tr -d 'nodejs' | sort -h -u)
+  local NODE_LATEST_VERSION=$(zypper -n search nodejs | egrep -i 'nodejs[0-9]{1,2} ' | cut -d'|' -f 2 | tr -d ' '| tr -d 'nodejs' | sort -h -u | tail -n1)
+
+  for v in $NODE_VERSIONS; do
+    echo "Found node version: $v"
+  done
+  echo "Will test only latest version: $NODE_LATEST_VERSION" 
+
+  # Install latest nodejs version and sources
+  zypper -n in --no-recommends "nodejs$NODE_LATEST_VERSION"
+  zypper -n si -D "nodejs$NODE_LATEST_VERSION"
+
+  test_node_version $NODE_LATEST_VERSION
+
+  # Print out info in case of skipped tests
+  if [ -s "$SKIPPED_TEST_LIST" ]; then
+    echo "Some tests have been skipped:"
+    cat $SKIPPED_TEST_LIST
+    echo ""
+    echo "Please look for 'SKIPPED' in the serial_terminal log for the full trace"
+  fi
+
+  # Fail if at least one test has failed. Print list of failed tests
+  if [ -s "$FAILED_TEST_LIST" ]; then
+    echo "Some tests have failed:"
+    cat $FAILED_TEST_LIST
+    echo ""
+    echo "Please look for 'FAILED' in the serial_terminal log for the full trace"
+    exit 1
+  else
+    echo "ALL tests have passed."
+  fi
+}
+
+# Hashmap of special flags required by some tests
+# Usage:   ["$VERSION $FILE"]="FLAG"
+# Example: ["10 test/directory/mytest.js"]="--myFlag"
+declare -A node_flags
+node_flags=(
+  ["10 test/parallel/test-tls-cli-max-version-1.2.js"]="--tls-max-v1.2"
+  ["10 test/parallel/test-tls-cli-min-version-1.0.js"]="--tls-min-v1.0 --tls-min-v1.1"
+  ["10 test/parallel/test-tls-cli-min-version-1.1.js"]="--tls-min-v1.1"
+  ["10 test/parallel/test-tls-cli-min-version-1.2.js"]="--tls-min-v1.2"
+  ["10 test/parallel/test-tls-cnnic-whitelist.js"]="--use-bundled-ca"
+  ["10 test/parallel/test-tls-dhe.js"]="--no-warnings"
+  ["10 test/parallel/test-tls-legacy-deprecated.js"]="--no-warnings"
+  ["10 test/parallel/test-tls-securepair-leak.js"]="--no-deprecation"
+  ["10 test/parallel/test-crypto-dh-leak.js"]="--noconcurrent-recompilation"
+)
+
+# Hashmap of tests that need to be skipped for a specific SLE version
+# Usage:   ["$VERSION $FILE" "$OS_VERSION"]="skip"
+# Example: ["14 tests/directory/mytest.js SLE_15"]="skip"
+declare -A skip_test
+skip_test=(
```

Comment:
I would suggest that the version stored here is the full version of the installed node version,

`rpm -q nodejs$(VERSION) --qf '%{version}-%{release}'`

And the test is skipped if the version match. When the version is updated, these tests should no longer fail.